### PR TITLE
implemented a few control msgs

### PR DESCRIPTION
**Pull request by damanmkumar**
The following control msgs have been implemented:
* host identification
* file read/write request
* chunk read/write request
* message type identification

Misc:
* fixed a bug in `echod`
* updated CMake to find SQLite3 reliably on all platforms
* removed unnecessary `#include` statements from `sadmd.cpp`

### CODE REVIEW

**Comment by damanmkumar** [comment]
partially implements #38

File: README.md [hunk 1]
```diff
@@ -10,6 +10,16 @@ $ cmake ..
 $ make
 ```
 
+To debug with GDB, replace `cmake ..` above with:
```
Comment (abmono93) [inline README.md:13]: Does this mean we can remove the `add_compile_options(ggdb)` in CMakeLists ?

File: src/comm/messages.cpp [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	return serialize(id, out) && serialize(protobuf, out);
+}
+
+template <typename Protobuf>
+bool
+recv(gpio::ZeroCopyInputStream* in, Protobuf& protobuf)
+{
+	return deserialize(&protobuf, in, nullptr);
+}
+
```
Comment (shu-g) [inline src/comm/messages.cpp]: This is really cool!

**Review by shu-g — APPROVED** [review]
Nice changes

**Review by abmono93 — APPROVED** [review]

**Pull request merged**